File: schemaorggo/broadcastevent.go
package schemaorggo

import "encoding/json"

// BroadcastEvent see : https://schema.org/BroadcastEvent
type BroadcastEvent struct {
	PublicationEvent

	typeContext

	// BroadcastOfEvent see : https://schema.org/broadcastOfEvent
	// The event being broadcast such as a sporting event or awards ceremony.
	// types : Event
	BroadcastOfEvent []*Event `json:"broadcastOfEvent,omitempty"`

	// IsLiveBroadcast see : https://schema.org/isLiveBroadcast
	// True is the broadcast is of a live event.
	// types : Boolean
	IsLiveBroadcast []bool `json:"isLiveBroadcast,omitempty"`

	// VideoFormat see : https://schema.org/videoFormat
	// The type of screening or video broadcast used (e.g. IMAX, 3D, SD, HD, etc.).
	// types : Text
	VideoFormat []string `json:"videoFormat,omitempty"`
}

func (v BroadcastEvent) intoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.PublicationEvent.intoMap(intop)

	into := *intop

	{
		var value interface{} = v.BroadcastOfEvent
		if len(v.BroadcastOfEvent) == 1 {
			value = v.BroadcastOfEvent[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["broadcastOfEvent"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.IsLiveBroadcast
		if len(v.IsLiveBroadcast) == 1 {
			value = v.IsLiveBroadcast[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["isLiveBroadcast"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.VideoFormat
		if len(v.VideoFormat) == 1 {
			value = v.VideoFormat[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["videoFormat"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v BroadcastEvent) asMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.intoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "BroadcastEvent"

	return data, nil
}

func (v BroadcastEvent) MarshalJSON() ([]byte, error) {
	data, err := v.asMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/broadcast_event.php
<?php

class BroadcastEvent extends PublicationEvent implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'BroadcastEvent';
	
	/**
	 * The event being broadcast such as a sporting event or awards ceremony.
	 * see : https://schema.org/broadcastOfEvent
	 * @var \Event|\Event[]
	 */
	public var $broadcast_of_event;
	
	/**
	 * True is the broadcast is of a live event.
	 * see : https://schema.org/isLiveBroadcast
	 * @var boolean|boolean[]
	 */
	public var $is_live_broadcast;
	
	/**
	 * The type of screening or video broadcast used (e.g. IMAX, 3D, SD, HD, etc.).
	 * see : https://schema.org/videoFormat
	 * @var string|string[]
	 */
	public var $video_format;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'BroadcastEvent'
		);
		
		$serialized = so_json_serialize( $this->broadcast_of_event );
		if ( ! empty( $serialized ) ) {
			$out['broadcastOfEvent'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->is_live_broadcast );
		if ( ! empty( $serialized ) ) {
			$out['isLiveBroadcast'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->video_format );
		if ( ! empty( $serialized ) ) {
			$out['videoFormat'] = $serialized;
		}
		
		return $out;
	}
}
